libkbfs/block_journal.go

```diff
@@ -614,7 +621,22 @@ func (j *blockJournal) removeBlockData(id BlockID) error {
 		return fmt.Errorf(
 			"Trying to remove data for referenced block %s", id)
 	}
-	return os.RemoveAll(j.blockPath(id))
+	path := j.blockPath(id)
+	err := os.RemoveAll(path)
+	if err != nil {
+		return err
+	}
+
+	// Remove the parent (splayed) directory if it's empty.
+	err = os.Remove(filepath.Dir(path))
+	// syscall.ENOTEMPTY is returned if the parent directory is
+	// not empty, which (as of go 1.7) IsExists returns true for.
+	//
+	// TODO: Verify that this behavior is the same on Windows.
```

Comment:
@taruti if you have time, mind seeing what error go returns on windows when retrying to remove a non-empty dir, and seeing if that passes os.IsExist?